src/classes/ProfileResults.ts

```diff
@@ -126,11 +114,16 @@ module.exports = class ProfileResults {
    * @param {String=} options.graph
    * @param {String=} options.graph_path
    */
-  compareToSnapshot (path, options = {}) {
-    let obj;
+  public compareToSnapshot(path: string, options: {
+    logResultsDiff?: boolean,
+    graph?: string,
+    graph_path?: string
+  } = {}): Comparison | null {
+    let obj: ResultData;
 
     try {
-      obj = JSON.parse(fs.readFileSync(path));
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+      obj = JSON.parse(fs.readFileSync(path).toString());
```

Comment:
Rather than disable the error, you should cast the result to the type it really is. `as ResultData`.